clang/lib/Driver/ToolChains/HIPAMD.cpp

```diff
@@ -283,7 +283,8 @@ void HIPAMDToolChain::addClangTargetOptions(
           DeviceOffloadingKind == Action::OFK_SYCL) &&
          "Only HIP and SYCL offloading kinds are supported for GPUs.");
 
-  CC1Args.push_back("-fcuda-is-device");
+  if (DeviceOffloadingKind != Action::OFK_SYCL)
```

Comment:
Yes, I will circle back to this after confirming whether that's the appropriate approach first. If so, the comment will be a close match to the previous change from the Cuda toolchain. I'll ping later, thanks!

Reply:
@srividya-sundaram 
I have updated the logic here. We'll keep `fcuda-is-device` (as it has been the case since the beginning of the HIPAMD plugin of DPCPP) and instead avoid setting `__CUDA_ARCH__` macro in the front-end in `InitPreprocessor.cpp` (see latest changes). The reasoning behind is that we don't know the impact of not passing `-fcuda-is-device` because all of the HIPAMD code seems to share a fair few things (including this option) with Cuda across clang. I have added comments where applicable and updated the driver test accordingly to not check the inexistence of `-fcuda-is-device` for AMDGPU targets in `-fsycl` mode.

Reply:
This also simplifies the Driver change now by minimising it to just the arch macro test. The rest is FE.